cql3/restrictions/statement_restrictions.hh

```diff
@@ -121,6 +121,11 @@ private:
     /// 5. if multi-column, then each element is a binary_operator
     std::vector<expr::expression> _clustering_prefix_restrictions;
 
+    /// Like _clustering_prefix_restrictions, but for the indexing table (if this is an index-reading statement).
+    ///
+    /// Elements are single-column binary operators.  The first element's RHS is a dummy value.
+    std::optional<std::vector<expr::expression>> _idx_tbl_ck_prefix;
```

Comment:
II{R,U}C, this the first element is the token, then comes the base table's partition key, then the base table's clustering key. Is this right? Either way, some documentation here will help.

Maybe we can use _clustering_prefix_restrictions, if they are never both used, with careful explanation.

Reply:
Ah, but for a local index it's a little different.

Reply:
> II{R,U}C, this the first element is the token, then comes the base table's partition key, then the base table's clustering key. Is this right?

Right, for a global index.

> Either way, some documentation here will help.

I don't want non-canonical documentation that repeats what's already documented in the index-creation code.  Redundant work, stale-bait, and all that...

Reply:
Then a link to the right place. It's just asking for people to get confused.

Reply:
Apparently, I'm not as averse to explaining this point as I pretend to be:
https://github.com/scylladb/scylla/blob/7a5111c580d97935374c55838182a4261b4c8f6a/cql3/restrictions/statement_restrictions.cc#L1286

I'll repeat the same thing here.  I'll also double-check that need_filtering() is correct for a local index, which its comment doesn't mention.